### PR TITLE
Separate tabs for Errors and Warnings in error overlay

### PR DESCRIPTION
**Pull request by ForsakenHarmony**

### CODE REVIEW

**Comment by ForsakenHarmony** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4001** <a href="https://app.graphite.dev/github/pr/vercel/turbo/4001" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/turbo/4001?utm_source=stack-comment).

File: crates/next-core/js/src/overlay/internal/container/Errors.tsx [hunk 1]
```diff
@@ -166,41 +166,58 @@ function useDisplayState(
...
   // TODO for now it's already closable, but in future we might want to block users from using a broken app
-  // const isClosable = !isLoading && !hasIssueWithError && !hasServerError;
+  // const isClosable = !isLoading && !hasTurbopackErrors && !hasServerError;
   const isClosable = true;
 
   const defaultTab =
-    hasIssueWithError || !hasErrors
-      ? TabId.TurbopackIssues
-      : TabId.RuntimeErrors;
+    TAB_PRIORITY.filter(
```
Comment (sokra) [inline crates/next-core/js/src/overlay/internal/container/Errors.tsx:203]: [Array.prototype.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) might be more suitable

File: crates/next-core/js/src/overlay/internal/container/Errors.tsx [hunk 2]
```diff
@@ -166,41 +166,58 @@ function useDisplayState(
...
+      (tabId) =>
+        ({
+          [TabId.TurbopackErrors]: turbopackErrors.length > 0,
+          [TabId.TurbopackWarnings]: turbopackWarnings.length > 0,
+          [TabId.RuntimeErrors]: hasErrors,
+        }[tabId])
+    )[0] ?? TabId.RuntimeErrors;
```
Comment (sokra) [inline crates/next-core/js/src/overlay/internal/container/Errors.tsx:210]:
```suggestion
    )[0];
```

File: crates/next-core/js/src/overlay/internal/container/Errors.tsx [hunk 3]
```diff
@@ -166,41 +166,58 @@ function useDisplayState(
...
 
   const [selectedTab, setSelectedTab] = React.useState<string>(defaultTab);
 
   React.useEffect(() => {
-    if (defaultTab === TabId.TurbopackIssues) {
-      setSelectedTab(TabId.TurbopackIssues);
+    if (defaultTab === TabId.TurbopackErrors) {
+      setSelectedTab(TabId.TurbopackErrors);
```
Comment (sokra) [inline crates/next-core/js/src/overlay/internal/container/Errors.tsx:216]:
```suggestion
    if (defaultTab) {
      setSelectedTab(defaultTab );
```

File: crates/next-core/js/src/overlay/internal/container/Errors.tsx (hunk 2 above)
Comment (ForsakenHarmony) [reply crates/next-core/js/src/overlay/internal/container/Errors.tsx:210]: didn't want it to be nullable

File: crates/next-core/js/src/overlay/internal/container/Errors.tsx (hunk 3 above)
Comment (ForsakenHarmony) [reply crates/next-core/js/src/overlay/internal/container/Errors.tsx:216]: This is here to switch to the turbopack error tab if those appear later on, but I'm not sure if it's a good idea to have

File: crates/next-core/js/src/overlay/internal/container/Errors.tsx (hunk 1 above)
Comment (ForsakenHarmony) [reply crates/next-core/js/src/overlay/internal/container/Errors.tsx]: true, changed it

**Pull request merged**